kvdb/cache.go

```diff
@@ -0,0 +1,930 @@
+package kvdb
+
+import (
+	"container/list"
+	"fmt"
+	"io"
+	"sync"
+
+	"github.com/btcsuite/btcwallet/walletdb"
+	"github.com/google/btree"
+)
+
+const (
+	treeDeg = 3
+)
+
+type cacheBucket struct {
+	seq  *uint64
```

Comment:
Any reason this is a pointer vs the raw value?

Reply:
This is to lazy initialize the `seq` (no need to fetch it for all buckets when prefetching). Fortunately `Sequence` is only part of the `ReadWriteBucket` interface so we can do this.